test/allowlist_for_publicAPI.json

```diff
@@ -2402,6 +2402,7 @@
     "List",
     "Optional",
     "TypeVar",
+    "Union",
```

Comment:
We should not add new entries to this file.
Adding the proper `__all__` to the submodule should do the trick here.

Reply:
Got it -- I misunderstood the purpose of this file